### PR TITLE
Use stdlib CLI

### LINKED ISSUE #28: Add "extract" command
As part of automating the release process for Pony projects, it would be helpful it I could run a "print" or "extract" command that would print out the changelog entries for the given version.

So for ponyc:

`changelog print 0.29.0`

would result in

```
### Fixed

- Do not permit leading zeros in JSON numbers ([PR #3167](https://github.com/ponylang/ponyc/pull/3167))
- Make reading via TCPConnection re-entrant safe ([PR #3175](https://github.com/ponylang/ponyc/pull/3175))
- Cleanup TCPConnection GC-safety mechanism for writev buffers ([PR #3177](https://github.com/ponylang/ponyc/pull/3177))
- Add SSL tests and fix some SSL related bugs ([PR #3174](https://github.com/ponylang/ponyc/pull/3174))
- Fix lib/llvm to support MacOS ([PR #3181](https://github.com/ponylang/ponyc/pull/3181))
- Close Denial of Service issue with TCPConnection.expect ([PR #3197](https://github.com/ponylang/ponyc/pull/3197))
- Fix return type checking to allow aliasing for non-ephemeral return types. ([PR #3201](https://github.com/ponylang/ponyc/pull/3201))

### Added

- Allow use of OpenSSL 1.1.1 when building Pony ([PR #3156](https://github.com/ponylang/ponyc/pull/3156))
- Add `pointer.offset` to get arbitrary pointer tags via offset ([PR #3177](https://github.com/ponylang/ponyc/pull/3177))
- Add DTrace/SystemTap probes for muted & unmuted events ([PR #3196](https://github.com/ponylang/ponyc/pull/3196))
- Add method to AsioEvent to see if an event is a oneshot ([PR #3198](https://github.com/ponylang/ponyc/pull/3198))

### Changed

- Do not permit leading zeros in JSON numbers ([PR #3167](https://github.com/ponylang/ponyc/pull/3167))
- Make TCPConnection yield on writes to not hog cpu ([PR #3176](https://github.com/ponylang/ponyc/pull/3176))
- Change how TCP connection reads data to improve performance ([PR #3178](https://github.com/ponylang/ponyc/pull/3178))
- Allow use of runtime TCP connect without ASIO one shot ([PR #3171](https://github.com/ponylang/ponyc/pull/3171))
- Simplify buffering in TCPConnection ([PR #3185](https://github.com/ponylang/ponyc/pull/3185))
- Allow for fine-grained control of yielding CPU in TCPConnection ([PR #3197](https://github.com/ponylang/ponyc/pull/3187))
- Make TCPConnection.expect partial ([PR #3197](https://github.com/ponylang/ponyc/pull/3197))
```

### LINKED ISSUE #30: Add "new" command
To make using the CHANGELOG tool easier. It would be good if there was a "new" command that would create an initial, empty, valid, changelog.

### PR DESCRIPTION
**Pull request by Theodus**
The changelog tool now uses the stdlib cli package. The `version`, `new`, and `get` commands have been added.

Closes #28, #30

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,9 +9,13 @@ All notable changes to the Pony compiler and standard library will be documented
 
 ### Added
 
+- Version command to print version
+
```
Comment (SeanTAllen) [inline CHANGELOG.md]: for the future, there shouldn't be a newline here, i will remove.

File: main.pony [hunk 2]
```diff
@@ -1,83 +1,185 @@
...
-use "itertools"
+use peg = "peg"
+use "time"
 
-actor Main
-  let help_text: String =
-    """
-    changelog-tool COMMAND <changelog file> [...]
+primitive Info
+  fun version(): String => "0.2.2"
```
Comment (SeanTAllen) [inline main.pony]: I'm going to end up changing this to using a VERSION doc and a -D to set it.

File: README.md [hunk 3]
```diff
@@ -44,7 +48,7 @@ All notable changes to the Pony compiler and standard library will be documented
 ```
 
 ```bash
-changelog-tool release CHANGELOG.md 0.13.1
+changelog-tool release 0.13.1
```
Comment (SeanTAllen) [inline README.md]: This seems like a breaking change to me, do I have to go to every place that `release` is used and change it?

It seems like now I need to use `-f` to set the file.

That would definitely make this a 0.3.0 version and I'll need to update every usage of changelog tool for releasing.

File: CHANGELOG.md [hunk 4]
```diff
@@ -9,9 +9,13 @@ All notable changes to the Pony compiler and standard library will be documented
...
+- New command for creating empty changelogs
 
 ### Changed
 
+- The changelog file name has become an optional argument
```
Comment (SeanTAllen) [inline CHANGELOG.md]: This seems to be a breaking change. "optional" doesn't quite capture the level of change this seems to be. can you verify that is correct?

File: README.md (hunk 3 above)
Comment (SeanTAllen) [reply README.md]: Yeah this is 100% a breaking change, I tested it. There are multiple breaking changes.

File: CHANGELOG.md (hunk 1 above)
Comment (SeanTAllen) [reply CHANGELOG.md]: can you remove the newline?

File: main.pony (hunk 2 above)
Comment (SeanTAllen) [reply main.pony]: can you change this to 0.3.0 given the breaking change?

**Comment by SeanTAllen** [comment]
I think the changelog needs to have a changed entry for the switch to CLI and it made clear from that that all the old commands are possibly broken.

**Comment by SeanTAllen** [comment]
@Theodus it appears that "unreleased" still has CHANGELOG.md as a required argument, is that the case?

File: README.md [hunk 5]
```diff
@@ -11,21 +11,25 @@ make
...
+changelog-tool verify
 ```
 ```
-verifying CHANGELOG.md...
-CHANGELOG.md is a valid changelog
+CHANGELOG.md is a valid changelog.
 ```
 
 ## Prepare a Changelog for a Release
 CHANGELOG.md:
```
Comment (SeanTAllen) [inline README.md]: Unrelated to this PR but, the CHANGELOG here is weird. It buries the "prepare for release" command. I think we should remove/move it.

Thoughts?

Comment (Theodus) [reply README.md]: Agreed. Is the updated README more clear?

**Comment by SeanTAllen** [comment]
I missed that you updated this.

**Pull request merged**